### PR TITLE
CloudFormation file types

### PR DESCRIPTION
**Pull request by abrooksv**
Due to new APIs being used, 192 builds will fail

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/clouddebug/DebuggerSupport.kt [hunk 1]
```diff
@@ -121,9 +121,16 @@ abstract class DebuggerSupport {
         val EP_NAME = ExtensionPointName<DebuggerSupport>("aws.toolkit.clouddebug.debuggerSupport")
 
         @JvmStatic
-        fun debuggers(): EnumMap<CloudDebuggingPlatform, DebuggerSupport> = EnumMap(EP_NAME.extensions.map {
-            it.platform to it
-        }.toMap())
+        fun debuggers(): EnumMap<CloudDebuggingPlatform, DebuggerSupport> {
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/clouddebug/DebuggerSupport.kt:124]: what the heck is this about?

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/clouddebug/DebuggerSupport.kt:124]: Its the earlier commits from amster getting in... the feature branch was made 2 commits behind master remotely

Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/clouddebug/DebuggerSupport.kt:124]: Messed up merge, will be fixed

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/CfnFileTypeDetector.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        if (file.extension != "template") {
+            return null
+        }
+
+        if (SERVERLESS_TRANSFORM in firstCharsIfText || CFN_FORMAT_VERSION in firstCharsIfText) {
+            val firstChar = firstCharsIfText.first { !it.isWhitespace() }
+
+            // Note: We find by name in case it is not loaded due to upstream plugin (yaml/json) is not enabled
+            val registry = FileTypeRegistry.getInstance()
+            return if (firstChar == '{') {
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/CfnFileTypeDetector.kt:30]: I know this was the existing detection code but `[`, is also valid JSON (despite being invalid CFN template) so maybe we should do the same check we do in VSTS https://github.com/aws/aws-vsts-tools/pull/332/files#diff-410117ffb18d53f752661b2ac5e3e65cR62

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/CfnFileTypeDetector.kt:30]: But very invalid CloudFrormation, its not meant to identify JSON or not, its to identify if its YAML CloudFromation or JSON Cloudformation

Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/CfnFileTypeDetector.kt:30]: Fair, makes sense

**Pull request merged**